src/flask_pyoidc/provider_configuration.py

```diff
@@ -57,19 +58,35 @@ def to_dict(self):
 
 
 class ProviderMetadata(OIDCData):
-    def __init__(self, issuer=None, authorization_endpoint=None, jwks_uri=None, **kwargs):
-        """
-        Args:
-            issuer (str): OP Issuer Identifier.
-            authorization_endpoint (str): URL of the OP's Authorization Endpoint
-            jwks_uri (str): URL of the OP's JSON Web Key Set
-            kwargs (dict): key-value pairs corresponding to
-                `OpenID Provider Metadata`_
-
-        .. _OpenID Provider Metadata:
-            https://openid.net/specs/openid-connect-discovery-1_0.html#ProviderMetadata
+
+    def __init__(self, issuer=None, authorization_endpoint=None, jwks_uri=None,
+                 token_endpoint=None, userinfo_endpoint=None,
+                 introspection_endpoint=None, registration_endpoint=None, **kwargs):
+        """OpenID Providers have metadata describing their configuration.
+
+        Parameters
+        ----------
+        issuer: str, Optional
```

Comment:
All optional types should be annotated `Optional[<type>]` to be consistent.

Reply:
Numpy docstring formatting rules are like this only.